analytics.go

```diff
@@ -310,8 +305,11 @@ func (c *Client) report(res *http.Response) {
 
 // Batch loop.
 func (c *Client) loop() {
+	defer close(c.shutdown)
+
 	var msgs []interface{}
-	tick := time.NewTicker(c.Interval)
+	var tick = time.NewTicker(c.Interval)
+	defer tick.Stop()
```

Comment:
This is just a safety measure, the current code flow didn't look like it had an issue but it always safer to clean resources in defer functions... just in case we things start to go wrong we don't happen to leak resources on top of that.